From 908b2184c3811a6f45782d400a5f7013fb803f62 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Wed, 10 May 2023 07:14:59 -0700
Subject: [PATCH] CMakeLists changes for OE builds

Signed-off-by: Matt Madison <matt@madison.systems>
---
 CMakeLists.txt | 38 +++++++++++++++++++-------------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8144450..b797eb6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required (VERSION 2.8)
 project (rkcrypto C)
 
+include(GNUInstallDirs)
+
 ################################################################################
 # toolchain config
 ################################################################################
@@ -37,35 +39,35 @@ set(RKCRYPTO_PUBLIC_HEADER
 	include/rkcrypto_common.h
 	include/rkcrypto_core.h
 	include/rkcrypto_mem.h
+	include/rkcrypto_random.h
+	include/rkcrypto_trace.h
 )
 
 include_directories(include)
 file(GLOB SOURCES "src/*.c")
 
-################################################################################
-# libdrm dependencies
-################################################################################
-include_directories(third_party/libdrm/include)
-include_directories(third_party/libdrm/include/drm)
-file(GLOB SOURCES_DRM "third_party/libdrm/src/*.c")
-add_definitions(-DMAJOR_IN_SYSMACROS=1 -D_GNU_SOURCE)
+set(RKCRYPTO_PRIVATE_HEADERS
+  include/dma-heap.h
+  include/rk_list.h
+  include/rkcrypto_core_int.h
+  include/cryptodev.h
+  include/rk_cryptodev.h
+)
 
+find_package(PkgConfig)
+pkg_check_modules(LIBDRM libdrm REQUIRED IMPORTED_TARGET libdrm)
 ################################################################################
-# libteec dependencies
-################################################################################
-add_definitions(-DBINARY_PREFIX=\"TEEC\")
-set(TEEC_PATH third_party/optee_client/libteec)
-include_directories(${TEEC_PATH}/../public ${TEEC_PATH}/include)
-file(GLOB SOURCES_TEEC
-	${TEEC_PATH}/src/tee_client_api.c
-	${TEEC_PATH}/src/teec_trace.c)
+
+find_library(LIBTEEC teec)
 
 ################################################################################
 # build librkcrypto shared library
 ################################################################################
 set(SHARED_LIB_NAME rkcrypto)
 
-add_library(${SHARED_LIB_NAME} SHARED ${SOURCES} ${SOURCES_DRM} ${SOURCES_TEEC})
+add_library(${SHARED_LIB_NAME} SHARED ${SOURCES} ${RKCRYPTO_PUBLIC_HEADER} ${RKCRYPTO_PRIVATE_HEADERS})
+target_link_libraries(${SHARED_LIB_NAME} PUBLIC ${LIBTEEC} PkgConfig::LIBDRM)
+set_target_properties(${SHARED_LIB_NAME} PROPERTIES VERSION 1.2.0 SOVERSION 1)
 set_target_properties(${SHARED_LIB_NAME} PROPERTIES PUBLIC_HEADER "${RKCRYPTO_PUBLIC_HEADER}")
 
 ################################################################################
@@ -85,6 +87,4 @@ add_subdirectory(test)
 ################################################################################
 install(TARGETS ${SHARED_LIB_NAME}
         LIBRARY DESTINATION "lib"
-        PUBLIC_HEADER DESTINATION "include")
-install(TARGETS ${STATIC_LIB_NAME}
-        ARCHIVE DESTINATION "lib")
+        PUBLIC_HEADER DESTINATION "include/librkcrypto")
-- 
2.34.1

