From 50efb12cc6130d2e43a510d21ef84e42ef2258ef Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Fri, 7 Nov 2025 12:49:51 -0800
Subject: [PATCH 3/3] Fix compilation error with g++ 13

Upstream-Status: Pending

Signed-off-by: Matt Madison <matt@madison.systems>
---
 rkaiq/ipc_server/MessageParser.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rkaiq/ipc_server/MessageParser.cpp b/rkaiq/ipc_server/MessageParser.cpp
index 8118da69..a1c34b94 100644
--- a/rkaiq/ipc_server/MessageParser.cpp
+++ b/rkaiq/ipc_server/MessageParser.cpp
@@ -19,6 +19,7 @@
 #define BUFFER_MAX_SIZE (1024 * 512)
 
 #ifndef LOGE
+#include <cstdio>
 #define LOGE printf
 #endif
 
-- 
2.43.0

